cpp/benchmarks/fixture/rmm_pool_raii.hpp

```diff
@@ -69,4 +69,15 @@ class rmm_pool_raii {
   std::shared_ptr<rmm::mr::device_memory_resource> mr;
 };
 
+/**
+ * Base fixture for cudf benchmarks using nvbench.
+ *
+ * Initializes the default memory resource to use the RMM pool device resource.
+ */
+struct nvbench_base_fixture {
+  rmm_pool_raii _mr;
+};
```

Comment:
Is it possible to inherit this from existing `NVBENCH_ENVIRONMENT` ?
This would allow global fixture to be layered.

Reply:
This would be a problem since the `NVBENCH_ENVIRONMENT` needs to be set before including `nvbench.cuh` so we'd get a circular dependency. I'd rather nvbench support layering first and then use that.
Just curious but what is the use case for layering?

Reply:
```suggestion
#ifndef NVBENCH_ENVIRONMENT
struct nvbench_base_fixture {
  rmm_pool_raii _mr;
};
#else
struct nvbench_base_fixture : NVBENCH_ENVIRONMENT {
  rmm_pool_raii _mr;
};
#undef NVBENCH_ENVIRONMENT
#endif
```

Reply:
I do not agree with this change for the reasons given previously.